src/nativeMain/kotlin/ByteOrder.kt

```diff
@@ -0,0 +1,25 @@
+package com.benasher44.uuid
+
+import kotlin.native.concurrent.SharedImmutable
+import kotlinx.cinterop.ByteVar
+import kotlinx.cinterop.IntVar
+import kotlinx.cinterop.alloc
+import kotlinx.cinterop.memScoped
+import kotlinx.cinterop.reinterpret
+import kotlinx.cinterop.value
+
+// @ExperimentalStdlibApi
+// @SinceKotlin("1.x")
+public actual enum class ByteOrder {
+    BIG_ENDIAN, LITTLE_ENDIAN;
+
+    public actual companion object {
+        @SharedImmutable
```

Comment:
not need: https://github.com/JetBrains/kotlin-native/blob/master/IMMUTABILITY.md

> To achieve mutable XOR global invariant, all globally visible state (currently, object singletons and enums) are automatically frozen.

Reply:
I took this directly from `kotlinx.io` assuming they know what they are doing: https://github.com/Kotlin/kotlinx-io/blob/ecf282cf2c7de2b06d3251f0081ffd90af15c814/core/nativeMain/src/kotlinx/io/NativeByteOrder.kt#L12-L19